compiler/rustc_builtin_macros/src/deriving/generic/mod.rs

```diff
@@ -995,7 +995,7 @@ impl<'a> MethodDef<'a> {
     /// `&self.x` because that might cause an unaligned ref. So for any trait
     /// method that takes a reference, if the struct impls `Copy` then we use a
     /// local block to force a copy:
-    /// ```
+    /// ```rust,ignore (pseudo-Rust)
```

Comment:
It can't compile.

```text
failures:

---- src/deriving/generic/mod.rs - deriving::generic::MethodDef<'a>::expand_struct_method_body (line 998) stdout ----
error: expected expression, found `==`
 --> src/deriving/generic/mod.rs:1003:20
  |
7 |         { self.x } == { other.y } && { self.y } == { other.y }
  |                    ^^ expected expression

error[E0404]: expected trait, found derive macro `Hash`
  --> src/deriving/generic/mod.rs:1006:6
   |
10 | impl Hash for A {
   |      ^^^^ not a trait
   |
help: consider importing one of these items instead
   |
2  | use core::hash::Hash;
   |
2  | use std::hash::Hash;
   |

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0404`.
Couldn't compile the test.
```

I didn't wanted to change the code, so I preferred to ignore it. I guess updating the code is better in this case.

In order to fix it, this diff has to be applied:
```diff
+++ b/compiler/rustc_builtin_macros/src/deriving/generic/mod.rs
@@ -995,12 +995,14 @@ fn create_method(
     /// `&self.x` because that might cause an unaligned ref. So for any trait
     /// method that takes a reference, if the struct impls `Copy` then we use a
     /// local block to force a copy:
-    /// ```rust,ignore (pseudo-Rust)
+    /// ```rust
+    /// use std::hash::Hash;
+    ///
     /// # struct A { x: u8, y: u8 }
     /// impl PartialEq for A {
     ///     fn eq(&self, other: &A) -> bool {
     ///         // Desugars to `{ self.x }.eq(&{ other.y }) && ...`
-    ///         { self.x } == { other.y } && { self.y } == { other.y }
+    ///         self.x == { other.y } && { self.y } == { other.y }
     ///     }
     /// }
     /// impl Hash for A {
```